### PR TITLE
I18n: POC for loading the messages depending on the user selected locale

### LINKED ISSUE #50538: Internationalisation: Load messages for user's selected locale

### PR DESCRIPTION
**Pull request by JoaoSilvaGrafana**
**What this PR does / why we need it**:

This proves that we can load just the messages for the user's locale.
We add `@lingui-loader` to be able to do this dynamically, as seen here: https://lingui.js.org/guides/dynamic-loading-catalogs.html

**Which issue(s) this PR fixes**:

Fixes #50538

**Special notes for your reviewer**:

Since we don't have the locale on the user preferences, for testing purposes I have used the "day of the week" preference 🙃 
Saturday -> Spanish
Sunday -> French
Monday/Default -> English

### CODE REVIEW

**Review by joshhunt — COMMENTED** [review]
Great stuff! I'm a _little bit_ concerned/curious about the loading/performance impact of (not tested it yet) loading messages dynamically after the main JS, but it's something we can iterate on!!

File: public/app/core/localisation.tsx [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 
-export function getI18n(locale = 'en') {
+export async function getI18n(locale = 'en') {
   if (i18nInstance) {
     return i18nInstance;
   }
-
-  i18n.load(locale, messages);
+  // Dynamically load the messages for the user's locale
+  const imp = await import(`@lingui/loader!../../locales/${locale}/messages.po`).catch((err) => {
```
[the remark this answers is not in the document]
Comment (JoaoSilvaGrafana) [reply public/app/core/localisation.tsx:14]: Yep you're totally right we don't need lingui-loader, it is better to have that extra step in the build process than adding this extra package and tie us to webpack. I'll remove it

**Comment by JoaoSilvaGrafana** [comment]
> Great stuff! I'm a _little bit_ concerned/curious about the loading/performance impact of (not tested it yet) loading messages dynamically after the main JS, but it's something we can iterate on!!

I did a quick performance test on the devtools for FCP and LCP and it didn't seem to have a big impact, but how do we normally test performance changes at Grafana? Do we have specific vitals we test?

File: public/app/core/localisation.tsx [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
   if (i18nInstance) {
     return i18nInstance;
   }
-
-  i18n.load(locale, messages);
+  // Dynamically load the messages for the user's locale
+  const imp = await import(`../../locales/${locale}/messages`).catch((err) => {
+    // TODO: Properly return an error if we can't find the messages for a locale
+    return err;
+  });
```
Comment (JoaoSilvaGrafana) [inline public/app/core/localisation.tsx:17]: @joshhunt Do you think this should also be behind the feature toggle in case it has some performance issues?

Comment (JoaoSilvaGrafana) [reply public/app/core/localisation.tsx:17]: Hmm is this for the case with the feature flag off? In that scenario I'm imagining we can just keep it always displaying english messages

Comment (joshhunt) [reply public/app/core/localisation.tsx]: Also makes sense :)

**Review by joshhunt — APPROVED** [review]
Great stuff!

**Pull request merged**